File: src/main/java/me/partlysanestudios/partlysaneskies/features/mining/WormWarning.java
//
// Written by Su386.
// See LICENSE for copyright and license notices.
//

package me.partlysanestudios.partlysaneskies.features.mining;

import me.partlysanestudios.partlysaneskies.PartlySaneSkies;
import me.partlysanestudios.partlysaneskies.render.gui.hud.BannerRenderer;
import me.partlysanestudios.partlysaneskies.render.gui.hud.PSSBanner;
import net.minecraftforge.client.event.ClientChatReceivedEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

public class WormWarning {

    String wormWarningString = "";
    long wormWarningBannerTime;

    @SubscribeEvent
    public void wormWarningChatEvent(ClientChatReceivedEvent event) {

        if (event.message.getUnformattedText().startsWith("You hear the sound of something approaching...")) {
            if (PartlySaneSkies.Companion.getConfig().mining.wormWarningBanner) {
                wormWarningBannerTime = PartlySaneSkies.Companion.getTime();
                wormWarningString = "A Worm Has Spawned!";

                BannerRenderer.INSTANCE.renderNewBanner(new PSSBanner(wormWarningString, (long) (PartlySaneSkies.Companion.getConfig().mining.wormWarningBannerTime * 1000), 3, PartlySaneSkies.Companion.getConfig().mining.wormWarningBannerColor.toJavaColor()));
            }
            if (PartlySaneSkies.Companion.getConfig().mining.wormWarningBannerSound) {
                PartlySaneSkies.Companion.getMinecraft().thePlayer.playSound("partlysaneskies:bell", 100, 1);
            }
        }
    }
}

File: src/main/kotlin/me/partlysanestudios/partlysaneskies/features/mining/crystalhollows/WormWarning.kt
//
// Written by Su386.
// See LICENSE for copyright and license notices.
//

package me.partlysanestudios.partlysaneskies.features.mining.crystalhollows

import me.partlysanestudios.partlysaneskies.PartlySaneSkies.Companion.config
import me.partlysanestudios.partlysaneskies.PartlySaneSkies.Companion.minecraft
import me.partlysanestudios.partlysaneskies.PartlySaneSkies.Companion.time
import me.partlysanestudios.partlysaneskies.events.SubscribePSSEvent
import me.partlysanestudios.partlysaneskies.events.minecraft.PSSChatEvent
import me.partlysanestudios.partlysaneskies.render.gui.hud.BannerRenderer.renderNewBanner
import me.partlysanestudios.partlysaneskies.render.gui.hud.PSSBanner

object WormWarning {
    private var wormWarningString = ""
    private var wormWarningBannerTime: Long = 0

    @SubscribePSSEvent
    fun onChat(event: PSSChatEvent) {
        if (event.component.unformattedText.startsWith("You hear the sound of something approaching...")) {
            if (config.wormWarningBanner) {
                wormWarningBannerTime = time
                wormWarningString = "A Worm Has Spawned!"
                renderNewBanner(
                    PSSBanner(
                        wormWarningString,
                        (config.wormWarningBannerTime * 1000).toLong(),
                        3f,
                        config.wormWarningBannerColor.toJavaColor(),
                    ),
                )
            }
            if (config.wormWarningBannerSound) {
                minecraft.thePlayer.playSound("partlysaneskies:bell", 100f, 1f)
            }
        }
    }
}
